Remove upstream's profile.release so we can have our own

--- Cargo.toml.orig	1970-01-01 00:00:01 UTC
+++ Cargo.toml
@@ -172,8 +172,3 @@ inherits = "dev"
 [profile.dev-rustls]
 inherits = "dev"
 
-[profile.release]
-opt-level = "z"
-lto = true
-codegen-units = 1
-panic = "abort"
